src/uwtools/rocoto.py

```diff
@@ -182,13 +182,60 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param config: Configuration data for this element.
         """
         e = SubElement(e, STR.dependency)
+        for key, block in config.items():
+            tag, _ = self._tag_name(key)
+            if tag in [STR.taskdep, STR.datadep, STR.timedep]:
+                self._add_task_dependency_operand(e, config={key: block})
+            elif tag in [STR.and_, STR.nand, STR.nor, STR.not_, STR.or_, STR.xor]:
+                self._add_task_dependency_operator(e, config={key: block})
+            elif tag in [STR.streq, STR.strneq]:
+                self._add_task_dependency_strequality(e, config={key: block})
+            else:
+                raise UWConfigError("Unhandled dependency type %s" % tag)
+
+    def _add_task_dependency_operand(self, e: Element, config: dict) -> None:
+        """
+        ???
+        """
         for key, block in config.items():
             tag, _ = self._tag_name(key)
             if tag == STR.taskdep:
                 self._set_attrs(SubElement(e, STR.taskdep), block)
+            elif tag == STR.datadep:
+                self._set_attrs(SubElement(e, STR.datadep), block)
+            else:
+                self._set_attrs(SubElement(e, STR.timedep), block)
```

Comment:
Is this whole `if` block equivalent to
```
self._set_attrs(SubElement(e, tag, block))
```
? We needed the `if` previously because we weren't ready to deal with all the possibilities, but aren't we now?

Reply:
I incorporated much of these changes/comments and simplified even further. I think it can still be breakdown more. I'm pushing what I currently have at the moment. Simplifying the methods down further leads me to edit the tests quite a bit and I wanted some opinions on which way to take those. I'll comment on that shortly as well.

Happy to take even more feedback since it'll be a continuous pairing down the methods/tests at this point. After adding much of your suggestions I was able to get back to 100% coverage as well. Thank you! @christinaholtNOAA @maddenp-noaa